packages/artifact/__tests__/artifact-http-client.test.ts

```diff
@@ -190,4 +190,51 @@ describe('artifact-http-client', () => {
     expect(mockHttpClient).toHaveBeenCalledTimes(1)
     expect(mockPost).toHaveBeenCalledTimes(1)
   })
+
+  it('should fail with a descriptive error', async () => {
+    // 409 duplicate error
+    const mockPost = jest.fn(() => {
+      const msgFailed = new http.IncomingMessage(new net.Socket())
+      msgFailed.statusCode = 409
+      msgFailed.statusMessage =
+        'Conflict: an artifact with this name already exists on the workflow run'
```

Comment:
_technically_ this `statusMessage` will be something like `OK` or `Not Found` from the headers

we'll never be changing this on the server response side

it would make more sense for this test to return a body with a `{"msg": "an artifact with this name already exists on the workflow run"}` to test the new logic we're adding 👍

Reply:
Gotcha! It should be `Conflict` to match the appropriate 409 error